lib/client.js

```diff
@@ -467,7 +467,14 @@ function Request(method, url) {
       self.emit('response', res);
     }
 
-    self.callback(err, res);
+    if (res && res.status >= 200 && res.status < 300) {
+      return self.callback(err, res);
+    }
+
+    var new_err = new Error('Unsucessful HTTP response');
```

Comment:
something like `'<code> "<status code string>"'`might help clarify that it's an http response error, for some of the terse ones like "Conflict"